Sources/hostmgr-helper/VirtualMachineSlot.swift

```diff
@@ -19,121 +19,160 @@ class VirtualMachineSlot: NSObject, ObservableObject {
 
     enum Status: Sendable {
         case empty
-        case starting(LaunchConfiguration)
-        case running(LaunchConfiguration, IPv4Address)
-        case stopping
+        case starting(LaunchConfiguration, Task<ManagedVirtualMachine, Error>)
+        case running(ManagedVirtualMachine)
+        case stopping(LaunchConfiguration)
         case crashed(Error)
     }
 
+    enum Errors: Error {
+        /// the slot was asked to start a new VM when it wasn't available
+        case invalidStartState
+        /// the VM was stopped while starting
+        case vmStartCancelled
```

Comment:
Nitpick

```suggestion
        /// The slot was asked to start a new VM when it wasn't available.
        case invalidStartState
        /// The VM was stopped while starting.
        case vmStartCancelled
```

Being documentation comments, they would look odd starting without an uppercase letter and missing a period. Or, at least they would look odd to sticklers such as myself 🤓

Reply:
Good call 👍